Automattic/woocommerce-payments
# Prevent infinite loop in usePaymentFailHandler effect

Fixes #

#### Changes proposed in this Pull Request
This PR fixes the unnecessary re-runs of the effect for handling failed payment due to [the non-specific dependencies list currently defined.](https://github.com/Automattic/woocommerce-payments/blob/develop/client/checkout/blocks/hooks.js#L58)









#### Testing instructions





* on `develop`, open the Blocks checkout and confirm that you see the `Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.` warnings 
* check out this branch and confirm no warning occurs anymore

-------------------


**Post merge**
